Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -277,7 +277,7 @@ class TabTrayController: UIViewController {
 
         tabDisplayManager.togglePrivateMode(isOn: !tabDisplayManager.isPrivate, createTabOnEmptyPrivateMode: false)
 
-        tabManager.willSwitchTabMode(leavingPBM: tabDisplayManager.isPrivate)
+        tabManager.willSwitchTabMode(leavingPBM: !tabDisplayManager.isPrivate)
```

Comment:
Instead, move this line before the above line (as that line modifies the value of `tabDisplayManager.isPrivate`) , so the `tabDisplayManager.isPrivate` will still be correct.

Reply:
Fixed.